RecoMuon/TrackerSeedGenerator/plugins/TSGForRoadSearch.cc

```diff
@@ -160,7 +160,7 @@ void TSGForRoadSearch::makeSeeds_0(const reco::Track & muon, std::vector<Traject
 
   //loop the parts until at least a compatible is found
   while (compatible.size()==0) {
-    switch ( inLayer->subDetector() ) {
+    switch ( GeomDetEnumerators::subDetGeom[inLayer->subDetector()] ) {
```

Comment:
While we're at it, after some private e-mails I'm no longer sure if this (use of `subDetGeom[]`) is the best solution. Although it solves the problem for Phase1 (`P1PXB` and `P1PXEC` get mapped to `PixelBarrel` and `PixelEndcap`), it also follows that the Phase2 enums get mapped by `P2PXEC -> PixelEndcap`, `P2OTB->TOB`, `P2OTEC->TID` which may be fine here or not (I have no idea). Alternative would be to add `P1PXB` and `P1PXEC` cases here, and leave the Phase2 cases to be dealt later (the two other switches in this file use LogError, so there is some hope to get notified from that).

I'd like to hear second opinion(s), and I'll leave the change as it is for now.